genie-agent/src/main/java/com/netflix/genie/agent/execution/statemachine/actions/SetUpJobAction.java

```diff
@@ -173,33 +192,26 @@ private void performJobSetup(final ExecutionContext executionContext) throws Set
         executionContext.setJobEnvironment(jobEnvironment);
     }
 
-    private File createJobDirectory(
-        final JobSpecification jobSpec
+    private void createJobDirectory(
+        final File jobDirectory
     ) throws SetUpJobException {
-        final File parentDir = jobSpec.getJobDirectoryLocation();
-        final String jobId = jobSpec.getJob().getId();
+        final File parentDir = jobDirectory.getParentFile();
```

Comment:
I'd recommend switching to the NIO library for a lot of this stuff as under the hood it leverages a lot of lower level OS optimizations for speed and arguably has a cleaner API. I used it for the Job File stuff.

Not necessary for this PR obviously but something to think about.